src/main/ApplicationImpl.cpp

```diff
@@ -177,6 +181,9 @@ ApplicationImpl::initialize(bool createNewDB)
     LedgerEntryIsValid::registerInvariant(*this);
     LiabilitiesMatchOffers::registerInvariant(*this);
     SponsorshipCountIsValid::registerInvariant(*this);
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+    OrderBookIsNotCrossed::registerInvariant(*this);
```

Comment:
as this invariant does not work in the general case (from what I understand), we should probably register it only where we need it

Reply:
I think I've managed that with the `registerAndEnableInvariant()` mechanism.